openedx/features/partners/forms.py

```diff
@@ -12,3 +25,129 @@ def clean_email(self):
         self.error_messages['unknown'] = _("We don't recognize the email: {}").format(self.cleaned_data["email"])
         return super(PartnerResetPasswordForm, self).clean_email()
 
+
+class UsernameField(student_forms.UsernameField):
+    """
+    Inheriting UsernameField from student app. Parent class is not validating username already exists
+    """
+
+    def clean(self, value):
+        clean_username = super(UsernameField, self).clean(value)
+
+        # check if user name is not already taken
+        conflicts = check_account_exists(username=clean_username)
+        if conflicts and 'username' in conflicts:
+            raise forms.ValidationError(
+                "The username you entered is already being used. Please enter another username.")
+
+        return clean_username
+
+
+class PartnerAccountCreationForm(forms.Form):
+    """
+    A form to validate account creation data for Give2Asia. It is currently only used for
+    validation, not rendering.
+    """
+
+    _NAME_MIN_LENGTH = 3
+    _EMAIL_INVALID_MSG = "A properly formatted e-mail is required"
+    _NAME_TOO_SHORT_MSG = "Your legal name must be a minimum of three characters long"
+
+    def __init__(self, data=None, tos_required=True):
+        super(PartnerAccountCreationForm, self).__init__(data)
+        if tos_required:
+            self.fields["terms_of_service"] = student_forms.TrueField(
+                error_messages={"required": "You must accept the terms of service."}
+            )
+
+    def validate_name(name):
```

Comment:
This is not a class method, just being used in the validator. So instead of adding `self` to it, I have moved this outside the class in the same file.